OrbitGl/App.cpp

```diff
@@ -1092,3 +1035,86 @@ void OrbitApp::LaunchRuleEditor(Function* a_Function) {
   m_RuleEditor->m_Window.Launch(a_Function);
   SendToUiNow("RuleEditor");
 }
+
+DataView* OrbitApp::CreateOrGetDataView(DataViewType type) {
+  switch (type) {
+    case DataViewType::FUNCTIONS:
+      if (!m_FunctionsDataView) {
+        m_FunctionsDataView = std::make_unique<FunctionsDataView>();
+        m_Panels.push_back(m_FunctionsDataView.get());
+      }
+      return m_FunctionsDataView.get();
+
+    case DataViewType::TYPES:
+      if (!m_TypesDataView) {
+        m_TypesDataView = std::make_unique<TypesDataView>();
+        m_Panels.push_back(m_TypesDataView.get());
+      }
+      return m_TypesDataView.get();
+
+    case DataViewType::LIVE_FUNCTIONS:
+      if (!m_LiveFunctionsDataView) {
+        m_LiveFunctionsDataView = std::make_unique<LiveFunctionsDataView>();
+        m_Panels.push_back(m_LiveFunctionsDataView.get());
+      }
+      return m_LiveFunctionsDataView.get();
+
+    case DataViewType::CALLSTACK:
+      if (!m_CallStackDataView) {
+        m_CallStackDataView = std::make_unique<CallStackDataView>();
+        m_Panels.push_back(m_CallStackDataView.get());
+      }
+      return m_CallStackDataView.get();
+
+    case DataViewType::GLOBALS:
+      if (!m_GlobalsDataView) {
+        m_GlobalsDataView = std::make_unique<GlobalsDataView>();
+        m_Panels.push_back(m_GlobalsDataView.get());
+      }
+      return m_GlobalsDataView.get();
+
+    case DataViewType::MODULES:
+      if (!m_ModulesDataView) {
+        m_ModulesDataView = std::make_unique<ModulesDataView>();
+        m_Panels.push_back(m_ModulesDataView.get());
+      }
+      return m_ModulesDataView.get();
+
+      /*
+      case DataViewType::SAMPLING:
+        if (!m_SamplingReportDataView) {
+          m_SamplingReportDataView = std::make_unique<SamplingReportDataView>();
+          m_Panels.push_back(m_SamplingReportDataView.get());
+        }
+        return m_SamplingReportDataView.get();
+      */
+
+    case DataViewType::PROCESSES:
+      if (!m_ProcessesDataView) {
+        m_ProcessesDataView = std::make_unique<ProcessesDataView>();
+        m_Panels.push_back(m_ProcessesDataView.get());
+        // TODO: Remove this after ProcessesDataView is untied from
+        // ModulesDataView
+        CreateOrGetDataView(DataViewType::MODULES);
+        m_ProcessesDataView->SetModulesDataView(m_ModulesDataView.get());
+      }
+      return m_ProcessesDataView.get();
+
+    case DataViewType::SESSIONS:
+      if (!m_SessionsDataView) {
+        m_SessionsDataView = std::make_unique<SessionsDataView>();
+        m_Panels.push_back(m_SessionsDataView.get());
+      }
+      return m_SessionsDataView.get();
+
+    case DataViewType::LOG:
+      if (!m_LogDataView) {
+        m_LogDataView = std::make_unique<LogDataView>();
+        m_Panels.push_back(m_LogDataView.get());
+      }
+      return m_LogDataView.get();
+
+    default:
+      FATAL("Unknown datatype=%d", static_cast<int>(type));
```

Comment:
I would assume the default is silencing the unhandled-enum-value-compiler-warning. Wouldn't it be better to put this code after the switch?